### ISSUE TITLE
Think about using a statically linked musl libc for release build

### ISSUE THREAD

**Issue by LukasKalbertodt** [elan-ev/tobira#116]
We encountered some problems when compiling Tobira on systems with a recent libc and then starting them on systems with an older libc. There are some ways around it, but one easy one would be to just build using musl as libc. This has lots of other benefits anyway. It is as easy as:

```
rustup target add x86_64-unknown-linux-musl
cargo build --target x86_64-unknown-linux-musl
```

This creates a completely statically linked binary that can be started on any linux system. From my tests, in the release build, this doesn't change the file size a lot (19MB for dynamically linked libc, 20MB for statically linked musl).

**Comment by ziegenberg** [comment]
Performance tests should be conducted to assure that the use of musl does not cause any performance degradation.

**Comment by LukasKalbertodt** [comment]
Absolutely. I plan on using jemalloc for Tobira, a user land allocator that is widely used and render the libc allocation performance pretty irrelevant. 

Also, we will only offer a musl build as alternative for at least quite some while. `x86_64-unknown-linux-gnu` builds won't go away anytime soon.

**Comment by LukasKalbertodt** [comment]
Just dropping two links here as "musl allocator super slow" might be outdated.

- Explanation of allocator rewrite in april 2020: https://twitter.com/RichFelker/status/1248486781785276419
- 1.2.1 seems to contain said alloc rewrite: https://musl.libc.org/releases.html

What I said in the previous comment still holds tho.
